src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1258 @@
+        {
+            char lhsString[256] = { 0 };
+            stringmgr::format_string(lhsString, lhs.name, (void*)&lhs.town);
+
+            char rhsString[256] = { 0 };
+            stringmgr::format_string(rhsString, rhs.name, (void*)&rhs.town);
+
+            return strcmp(lhsString, rhsString) < 0;
+        }
+
+        // 0x00457A9F
+        static bool orderByStatus(openloco::industry& lhs, openloco::industry& rhs)
+        {
+            char lhsString[256] = { 0 };
+            const char* lhsBuffer = stringmgr::get_string(string_ids::buffer_1250);
+            lhs.getStatusString((char*)lhsBuffer);
+
+            stringmgr::format_string(lhsString, string_ids::buffer_1250);
+
+            char rhsString[256] = { 0 };
+            const char* rhsBuffer = stringmgr::get_string(string_ids::buffer_1250);
+            rhs.getStatusString((char*)rhsBuffer);
+
+            stringmgr::format_string(rhsString, string_ids::buffer_1250);
+
+            return strcmp(lhsString, rhsString) < 0;
+        }
+
+        static uint8_t getAverageTransportedCargo(const openloco::industry& industry)
+        {
+            auto industryObj = objectmgr::get<industry_object>(industry.object_id);
+            uint8_t productionTransported = -1;
+
+            if (industryObj->producesCargo())
+            {
+                productionTransported = industry.produced_cargo_transported[0];
+                if (industryObj->produced_cargo_type[1] != 0xFF)
+                {
+                    productionTransported = industry.produced_cargo_transported[1];
+                    if (industryObj->produced_cargo_type[0] != 0xFF)
+                    {
+                        productionTransported += industry.produced_cargo_transported[0];
+                        productionTransported /= 2;
+                    }
+                }
+            }
+            return productionTransported;
+        }
+
+        // 0x00457AF3
+        static bool orderByProductionTransported(const openloco::industry& lhs, const openloco::industry& rhs)
+        {
+            auto lhsVar = getAverageTransportedCargo(lhs);
+
+            auto rhsVar = getAverageTransportedCargo(rhs);
+
+            return rhsVar < lhsVar;
+        }
+
+        // 0x00457A52, 0x00457A9F, 0x00457AF3
+        static bool getOrder(const SortMode mode, openloco::industry& lhs, openloco::industry& rhs)
+        {
+            switch (mode)
+            {
+                case SortMode::Name:
+                    return orderByName(lhs, rhs);
+
+                case SortMode::Status:
+                    return orderByStatus(lhs, rhs);
+
+                case SortMode::ProductionTransported:
+                    return orderByProductionTransported(lhs, rhs);
+            }
+
+            return false;
+        }
+
+        // 0x00457991
+        static void updateIndustryList(window* self)
+        {
+            auto edi = -1;
```

Comment:
edi could be called `chosenIndustry`. We want to avoid using the register names as much as possible.